include/deal.II/lac/scalapack.h

```diff
@@ -0,0 +1,478 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2017 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_scalapack_h
+#define dealii_scalapack_h
+
+#include <deal.II/base/config.h>
+
+#ifdef DEAL_II_WITH_SCALAPACK
+
+#include <deal.II/base/exceptions.h>
+#include <deal.II/lac/full_matrix.h>
+#include <deal.II/lac/lapack_support.h>
+#include <deal.II/base/mpi.h>
+#include <mpi.h>
+
+#include <memory>
+
+DEAL_II_NAMESPACE_OPEN
+
+
+/**
+ * Forward declaration of class ScaLAPACKMatrix
+ */
+template <typename NumberType> class ScaLAPACKMatrix;
+
+
+/**
+ * A class taking care of setting up the process grid for BLACS
+ * ScaLapack matrices will have shared pointer of this class to perform block-cyclic distribution
+ *
+ * @author Benjamin Brands, 2017
+ */
+class ProcessGrid
```

Comment:
do we need to expose this class to the user? If it is just an implementation detail, put it inside the .cc

Reply:
yes, we do because users need to use the same grid for multiple matrices in case of eigenvalue calculations. So far i think it's hidden, but potentially this is like a partitioner for parallel vector which we share.